util/hash_map.h

```diff
@@ -0,0 +1,57 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.  This
+//  source code is licensed under the BSD-style license found in the LICENSE
+//  file in the root directory of this source tree. An additional grant of
+//  patent rights can be found in the PATENTS file in the same directory.
+//
+
+#pragma once
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+namespace rocksdb {
+
+template <typename K, typename V, size_t size = 128>
```

Comment:
Let's add a comment section stating why we implemented our own HashMap (why it's better than unordered_map)